### PR TITLE
piksi_ins startup: update startup scripts to avoid resets in some con…

### PR DESCRIPTION
**Pull request by denniszollo**
This PR is to address two things

1. If someone or a Swifter manages to build an SDK version of piksi_buildroot and then flashes that to a Duro_Inertial, the device will end up in a restart loop that may even prevent them from doing anything
2. Sometimes Swifters ended up with the Smoothpose router config even though their device didn't have a license file or their build was missing the PoseDaemon.

I think I address either case with the changes.  I want to put it into v2.2.0 release and master with hopes that v2.2.0 will have SDK support that is reasonable.

TODO: test travis build thoroughly for all combinations of license and setting.

### CODE REVIEW

File: package/common_init/overlay/etc/init.d/common.sh [hunk 1]
```diff
@@ -171,8 +171,19 @@ lockdown()
...
+
+# in order to be a real piksi_ins, you need all of the following:
+#  - output mode configured to either "Debug" or "Loosely Coupled"
+#  - the license file must exist
+#  - the PoseDaemon executable must exist
+
+detect_piksi_ins()
+{
+  ins_output_mode=$(query_config --section ins --key output_mode)
+  { detect_piksi_ins_router || [[ "$ins_output_mode" == "Debug" ]];} && [[ -f /usr/bin/PoseDaemon ]] && [[ -f /persistent/licenses/smoothpose_license.json ]]
```
Comment (silverjam) [inline package/common_init/overlay/etc/init.d/common.sh]: Would appreciate some formatting to make this more readable:

```sh
   { detect_piksi_ins_router || [[ "$ins_output_mode" == "Debug" ]]; } \
     && [[ -f /usr/bin/PoseDaemon ]] \
     && [[ -f /persistent/licenses/smoothpose_license.json ]]
```

File: package/common_init/overlay/etc/init.d/S80endpoint_router_sbp [hunk 2]
```diff
@@ -16,7 +16,7 @@ setup_permissions()
 
 starling_daemon_enabled=$(query_config --section experimental_flags --key starling_on_linux)
 
-if detect_piksi_ins; then
+if detect_piksi_ins_router && detect_piksi_ins; then
```
Comment (silverjam) [inline package/common_init/overlay/etc/init.d/S80endpoint_router_sbp]: Seems like we're calling `detect_piksi_ins_router` twice with this, won't `detect_piksi_ins` also call `detect_piksi_ins_router` ?

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.11-3-g66d733c4
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=851293)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.11-3-g66d733c4&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.11-3-g66d733c4&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.11-3-g66d733c4
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.11-3-g66d733c4)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.11-3-g66d733c4/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.11-3-g66d733c4

File: package/common_init/overlay/etc/init.d/S80endpoint_router_sbp (hunk 2 above)
Comment (denniszollo) [reply package/common_init/overlay/etc/init.d/S80endpoint_router_sbp]: good point.

File: package/common_init/overlay/etc/init.d/common.sh (hunk 1 above)
Comment (denniszollo) [reply package/common_init/overlay/etc/init.d/common.sh]: Will do.

File: package/common_init/overlay/etc/init.d/S80endpoint_router_sbp (hunk 2 above)
Comment (denniszollo) [reply package/common_init/overlay/etc/init.d/S80endpoint_router_sbp]: we are calling this twice, but it's an AND condition for the router, while it is in an or condition in detect_piksi_ins.  In order to use the smoothpose router, we need to be a real_piksi_ins and have config that requires smoothpose routing.

Comment (denniszollo) [reply package/common_init/overlay/etc/init.d/S80endpoint_router_sbp]: In seeing that, I think I can make the logic less ugly and clearer though.  I will think for second and improve.

Comment (denniszollo) [reply package/common_init/overlay/etc/init.d/S80endpoint_router_sbp]: I thought about it and I think this is still the clearest

Comment (silverjam) [reply package/common_init/overlay/etc/init.d/S80endpoint_router_sbp]: ok

**Comment by denniszollo** [comment]
The only errors were Nap verification failed:
https://gnss-analysis.swiftnav.com/test_result/workflowId=7321bd2d-0c0f-4fc7-9993-6d7a2ad092a2&deviceId=PKdc134fe9a4284506baa564b9be66c81c#logged_errors_host_time_utc_

**Pull request merged**